File: core/src/kr/edoli/imview/ImView.java
package kr.edoli.imview;

import com.badlogic.gdx.Game;
import com.badlogic.gdx.Gdx;
import kr.edoli.imview.ui.MainScreen;
import lombok.Data;

public class ImView extends Game {
	private long targetDelay = 1000 / 60;
	private long start;

	public static AppArgs args = new AppArgs();

	public ImView(String imagePath) {
		args.setImagePath(imagePath);
	}

	@Override
	public void create() {
		Context.imagePath = ImView.args.getImagePath();

		setScreen(new MainScreen());
		start = System.currentTimeMillis();
	}

	@Override
	public void render() {
		long diff = System.currentTimeMillis() - start;

		long delay = targetDelay - diff;

		if (delay > 0) {
			try {
				Thread.sleep(delay);
			} catch (InterruptedException e) {
				e.printStackTrace();
			}
		}

		if (screen != null) screen.render(Gdx.graphics.getDeltaTime());

		start = System.currentTimeMillis();
	}

	@Data
	public static class AppArgs {
		private String imagePath;
	}
}

File: core/src/kr/edoli/imview/ImView.kt
package kr.edoli.imview

import com.badlogic.gdx.Game
import com.badlogic.gdx.Gdx
import kr.edoli.imview.image.Test
import kr.edoli.imview.ui.screen.MainScreen

/**
 * Created by daniel on 16. 9. 10.
 */

class ImView : Game() {

    override fun create() {
        initPreference()


        setScreen(MainScreen())


        val runnable = object : Runnable {

            private var lastTime: Long = 0

            override fun run() {
                var currentTime = System.nanoTime()

                while (currentTime - lastTime < 16000000) {
                    try {
                        Thread.sleep(16)
                    } catch (e: InterruptedException) {
                        e.printStackTrace()
                    }
                    currentTime = System.nanoTime()
                }
                Gdx.app.postRunnable(this)

                lastTime = currentTime
            }
        }

        Gdx.app.postRunnable(runnable)
    }

    private fun initPreference() {
        val pref = Gdx.app.getPreferences("kr.edoli.imview.context")

        Context.isShowCrosshair.update(pref.getBoolean("isShowCrosshair", true))
        Context.isFixToolBar.update(pref.getBoolean("isFixToolBar", true))
        Context.isShowInfo.update(pref.getBoolean("isShowInfo", true))

        Context.isShowCrosshair.subscribe { pref.putBoolean("isShowCrosshair", it).flush() }
        Context.isFixToolBar.subscribe { pref.putBoolean("isFixToolBar", it).flush() }
        Context.isShowInfo.subscribe { pref.putBoolean("isShowInfo", it).flush() }
    }
}